src/pytds/__init__.py

```diff
@@ -223,6 +230,9 @@ def connect(
     login.cafile = cafile
     login.validate_host = validate_host
     login.enc_login_only = enc_login_only
+    if access_token_callable is not None:
+        login.access_token = access_token_callable()
```

Comment:
call to access_token_callable() should be done in _connect method, that is where actual connection attempt is made
also there you should keep track of expiration of the token, you should only call this method if token is expired or not yet created

Reply:
Ok, lets rely on `msal` to cache tokens, but this still needs to be moved to `_connect` method, since otherwise it will not refresh token when new connections are made on connection pool.